misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java

```diff
@@ -320,6 +308,27 @@ public static URI covertToBalaUri(URI fileUri) throws URISyntaxException, Illega
         throw new IllegalArgumentException("unsupported URI with scheme: " + fileUri.getScheme());
     }
 
+    /**
+     * Get the path from given string URI (If the given URI scheme is `bala`, it will still be converted to `file`
+     * scheme).
+     *
+     * @param fileUri file uri
+     * @return {@link Optional} Path from the URI
+     */
+    private static Optional<Path> getPathFromURI(String fileUri) {
+        try {
+            URI uri = URI.create(fileUri);
```

Comment:
is `fileUri` always a remote URL?

Reply:
nope in this context it'll always be a local file uri with different uri schemas (i.e. `file`, `bala`)